diff -ruN a/build.rs b/build.rs
--- a/build.rs
+++ b/build.rs
@@ -24,7 +24,7 @@
                     path.components().skip(2)
                         .map(|c| c.as_os_str().to_str().expect("valid UTF-8"))
                         .collect::<Vec<_>>().join("/"),
-                    cwd.join(path))?;
+                    path)?;
             } else if path.is_dir() {
                 dirs.push(path.clone());
             }
diff -ruN a/src/test_framework/data.rs b/src/test_framework/data.rs
--- a/src/test_framework/data.rs
+++ b/src/test_framework/data.rs
@@ -21,7 +21,7 @@
 
         macro_rules! add {
             ( $key: expr, $path: expr ) => {
-                m.insert($key, include_bytes!($path))
+                m.insert($key, include_bytes!(concat!(env!("CARGO_MANIFEST_DIR"), "/", $path)))
             }
         }
         include!(concat!(env!("OUT_DIR"), "/tests.index.rs.inc"));
