--- a/build.rs
+++ b/build.rs
@@ -11,7 +11,11 @@
 
 fn capnp(src: &str) {
     println!("rerun-if-changed={}", src);
-    ::capnpc::CompilerCommand::new().file(src).run().unwrap();
+    let mut cmd = ::capnpc::CompilerCommand::new();
+    if let Ok(capnp) = std::env::var("CAPNP") {
+        cmd.capnp_executable(capnp);
+    }
+    cmd.file(src).run().unwrap();
 }
 
 /// Builds the index of the test data for use with the `::tests`
